sys/include/net/gcoap.h

```diff
@@ -793,6 +796,25 @@ static inline ssize_t gcoap_request(coap_pkt_t *pdu, uint8_t *buf, size_t len,
                 : -1;
 }
 
+/**
+ * @brief   Sends a buffer containing a CoAP request to the provided endpoint
+ *          and reports back the gcoap_request_memo_t that was used
+ *
+ * @param[in] buf           Buffer containing the PDU
+ * @param[in] len           Length of the buffer
+ * @param[in] remote        Destination for the packet
+ * @param[in] report_memo   Pointer to the memo used for sending the request
+ * @param[in] resp_handler  Callback when response received, may be NULL
+ * @param[in] context       User defined context passed to the response handler
```

Comment:
As the memo is return, it might make sense not to have setting the context here in the first place (especially if any of the unions mentioned above is used).